Can't serialize image with non-zero offset

If an image has a non zero offset the python script node throws an Error:
```
java.lang.RuntimeException: Could not serialize image. Possible reasons: Unsupported image type, dimensionality of the image,...
	at org.knime.knip.serialization.ImgPlusToBytesConverter.serialize(ImgPlusToBytesConverter.java:82)
	at org.knime.knip.knimepython.ImgPlusSerializerFactory$1.serialize(ImgPlusSerializerFactory.java:31)
	at org.knime.knip.knimepython.ImgPlusSerializerFactory$1.serialize(ImgPlusSerializerFactory.java:1)
	at org.knime.python2.extensions.serializationlibrary.interfaces.impl.BufferedDataTableIterator.dataRowToRow(BufferedDataTableIterator.java:397)
	at org.knime.python2.extensions.serializationlibrary.interfaces.impl.BufferedDataTableIterator.next(BufferedDataTableIterator.java:154)
	at org.knime.python2.serde.flatbuffers.Flatbuffers.tableToBytes(Flatbuffers.java:226)
	at org.knime.python2.kernel.PythonKernel.putDataTable(PythonKernel.java:664)
	at org.knime.python2.kernel.PythonKernel.putDataTable(PythonKernel.java:698)
	at org.knime.python2.nodes.script.PythonScriptNodeModel.execute(PythonScriptNodeModel.java:87)
	at org.knime.core.node.NodeModel.execute(NodeModel.java:733)
	at org.knime.core.node.NodeModel.executeModel(NodeModel.java:567)
	at org.knime.core.node.Node.invokeFullyNodeModelExecute(Node.java:1172)
	at org.knime.core.node.Node.execute(Node.java:959)
	at org.knime.core.node.workflow.NativeNodeContainer.performExecuteNode(NativeNodeContainer.java:561)
	at org.knime.core.node.exec.LocalNodeExecutionJob.mainExecute(LocalNodeExecutionJob.java:95)
	at org.knime.core.node.workflow.NodeExecutionJob.internalRun(NodeExecutionJob.java:179)
	at org.knime.core.node.workflow.NodeExecutionJob.run(NodeExecutionJob.java:110)
	at org.knime.core.util.ThreadUtils$RunnableWithContextImpl.runWithContext(ThreadUtils.java:328)
	at org.knime.core.util.ThreadUtils$RunnableWithContext.run(ThreadUtils.java:204)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.knime.core.util.ThreadPool$MyFuture.run(ThreadPool.java:123)
	at org.knime.core.util.ThreadPool$Worker.run(ThreadPool.java:246)
Caused by: java.lang.ArrayIndexOutOfBoundsException: -3027
	at net.imglib2.img.basictypeaccess.array.AbstractByteArray.getValue(AbstractByteArray.java:61)
	at net.imglib2.type.numeric.integer.GenericByteType.getValue(GenericByteType.java:98)
	at net.imglib2.type.numeric.integer.UnsignedByteType.get(UnsignedByteType.java:159)
	at net.imglib2.type.numeric.integer.UnsignedByteType.getIntegerLong(UnsignedByteType.java:176)
	at net.imglib2.type.numeric.integer.AbstractIntegerType.getRealDouble(AbstractIntegerType.java:62)
	at io.scif.img.ImgSaver.getPlaneArray(ImgSaver.java:850)
	at io.scif.img.ImgSaver.writePlanes(ImgSaver.java:626)
	at io.scif.img.ImgSaver.writeImg(ImgSaver.java:542)
	at io.scif.img.ImgSaver.writeImg(ImgSaver.java:528)
	at io.scif.img.ImgSaver.writeImg(ImgSaver.java:462)
	at io.scif.img.ImgSaver.saveImg(ImgSaver.java:244)
	at io.scif.img.ImgSaver.saveImg(ImgSaver.java:223)
	at org.knime.knip.serialization.ImgPlusToBytesConverter.serialize(ImgPlusToBytesConverter.java:77)
	... 22 more
```

I created a workflow which illustrates the problem: [KNIP-Python-offset.zip](https://github.com/knime-ip/knip-python-extensions/files/1848146/KNIP-Python-offset.zip)
But the error only appears when the image cropper was already executed when the workflow is loaded. (This suggest that there is more to the issue than expected).

Also, see [this forumpost](https://forum.knime.com/t/segment-cropper-corrupts-images-for-python-nodes) for a workflow that always fails.

--- a/org.knime.knip.knimepython/src/org/knime/knip/serialization/ImgPlusToBytesConverter.java
+++ b/org.knime.knip.knimepython/src/org/knime/knip/serialization/ImgPlusToBytesConverter.java
@@ -26,7 +26,12 @@ import io.scif.io.RandomAccessOutputStream;
 import net.imagej.ImgPlus;
 import net.imagej.axis.Axes;
 import net.imagej.axis.CalibratedAxis;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.img.Img;
+import net.imglib2.img.ImgView;
+import net.imglib2.type.Type;
 import net.imglib2.util.Intervals;
+import net.imglib2.view.Views;
 
 /**
  * Serializing ImgPlus instances to byte stream. Used format is .tif.
@@ -37,7 +42,7 @@ import net.imglib2.util.Intervals;
  * @author Christian Dietz (University of Konstanz)
  *
  */
-public class ImgPlusToBytesConverter {
+public class ImgPlusToBytesConverter<T extends Type<T>> {
 
 	/**
 	 * ImgSaver to write ImgPlus to stream as tif
@@ -63,7 +68,8 @@ public class ImgPlusToBytesConverter {
 			m_writer = createWriter();
 		}
 
-		final ImgPlus<?> imgPlus = TypeUtils.converted(value.getImgPlus());
+		@SuppressWarnings("unchecked")
+		final ImgPlus<T> imgPlus = (ImgPlus<T>) TypeUtils.converted(value.getImgPlus());
 
 		try {
 			final ByteBuffer buffer = ByteBuffer.allocate((int) Intervals.numElements(value.getDimensions()));
@@ -74,7 +80,8 @@ public class ImgPlusToBytesConverter {
 			m_writer.getMetadata().setDatasetName("");
 			m_writer.setDest(new RandomAccessOutputStream(handle), 0);
 
-			m_saver.saveImg(m_writer, imgPlus.getImg(), m_scifioConfig);
+			final Img<T> zeroMinImg = ImgView.wrap(Views.zeroMin(imgPlus.getImg()), imgPlus.factory());
+			m_saver.saveImg(m_writer, zeroMinImg, m_scifioConfig);
 
 			m_writer.close();
 			return handle.getBytes();

